traffic_ops/testing/api/v4/serverupdatestatus_test.go

```diff
@@ -29,6 +29,119 @@ import (
+			if err != nil {
+				t.Errorf("cannot get Server by name '%s': %v - alerts: %+v", *testServer.HostName, err, resp.Alerts)
+			} else if len(resp.Response) != 1 {
+				t.Fatalf("GET Server expected 1, actual %v", len(resp.Response))
+			}
+
+			beforeServer := resp.Response[0]
+
+			// Ensure baseline
+			if beforeServer.UpdPending == nil {
+				t.Fatalf("Server '%s' had nil UpdPending before update status change", *testServer.HostName)
+			}
+			if beforeServer.RevalPending == nil {
+				t.Fatalf("Server '%s' had nil RevalPending before update status change", *testServer.HostName)
+			}
+			if beforeServer.ConfigUpdateTime == nil {
+				t.Fatalf("Server '%s' had nil ConfigUpdateTime before update status change", *testServer.HostName)
+			}
+			if beforeServer.ConfigApplyTime == nil {
+				t.Fatalf("Server '%s' had nil ConfigApplyTime before update status change", *testServer.HostName)
+			}
+			if beforeServer.RevalUpdateTime == nil {
+				t.Fatalf("Server '%s' had nil RevalUpdateTime before update status change", *testServer.HostName)
+			}
+			if beforeServer.RevalApplyTime == nil {
+				t.Fatalf("Server '%s' had nil RevalApplyTime before update status change", *testServer.HostName)
+			}
+
+			_, _, err = TOSession.CreateCDNLock(tc.CDNLock{CDN: *testServer.CDNName, Soft: util.BoolPtr(false)}, client.NewRequestOptions())
+			if err != nil {
+				t.Errorf("cannont acquire lock on the Server '%s' CDN '%s: %v", *testServer.HostName, *testServer.CDNName, err)
+			}
+
+			opsSession, _, err := client.LoginWithAgent(Config.TrafficOps.URL, Config.TrafficOps.Users.Operations,
+				Config.TrafficOps.UserPassword, true, "to-api-v4-locks-ops",
+				true, time.Second*time.Duration(Config.Default.Session.TimeoutInSecs))
+			if err != nil {
+				t.Errorf("cannot login as '%s' user: %v", Config.TrafficOps.Users.Operations, err)
+			}
+
+			badQueryOpts := client.NewRequestOptions()
+			badQueryOpts.QueryParameters.Set("updated", "true")
+			if _, _, err := opsSession.SetUpdateServerStatusTimes(*testServer.HostName, configApply, revalApply, badQueryOpts); err == nil {
+				t.Fatalf("expected SetUpdateServerStatusTimes to not allow updates due to `updated` field")
+			}
+
+			badQueryOpts = client.NewRequestOptions()
+			badQueryOpts.QueryParameters.Set("reval_updated", "true")
+			if _, _, err := opsSession.SetUpdateServerStatusTimes(*testServer.HostName, configApply, revalApply, badQueryOpts); err == nil {
+				t.Fatalf("expected SetUpdateServerStatusTimes to not allow updates due to `reval_updated` field")
+			}
+
+			if alerts, _, err := opsSession.SetUpdateServerStatusTimes(*testServer.HostName, configApply, revalApply, client.RequestOptions{}); err != nil {
+				t.Fatalf("SetUpdateServerStatusTimes error. expected: nil, actual: %v - alerts: %+v", err, alerts.Alerts)
+			}
+
+			resp, _, err = opsSession.GetServers(opts)
+			if err != nil {
+				t.Errorf("cannot GET Server by name '%s': %v - alerts: %+v", *testServer.HostName, err, resp.Alerts)
```

Comment:
This should be `fatal`, to prevent a panic in the following section, if `err` is not `nil` and `resp` is `nil`.